gcloud/pubsub/topic.py

```diff
@@ -113,6 +122,8 @@ def publish(self, message, **attrs):
         :rtype: str
         :returns: message ID assigned by the server to the published message
         """
+        if self.add_timestamp_to_messages:
+            attrs['timestamp'] = '%sZ' % _NOW().isoformat()
```

Comment:
We can add code (and extra tests) to handle that case, or just decide that if the user configured the topic with the flag, applying it blindly is the Right Thing (TM).  I had considered both, and decided to go for the simpler case.